### ISSUE TITLE
metadata: handle UDT types correctly

### ISSUE THREAD

**Issue by Zariel** [gocql/gocql#1053]
Currently UDT's will be parsed into NativeType{typ: TypeCustom} which is wrong, we should read the types from system_schemas.types and then create a UDTType from that.

**Comment by martin-sucha** [comment]
Currently `getTypeInfo` does not handle user-defined types. We'd need to pass information about all the types and recursively reference the other types (since user-defined types can contain other user-defined types).

**Comment by OleksiienkoMykyta** [comment]
gocql still can't handle UDT natively.
We should provide Fetching UDT definitions from system_schema.types, as Martin mentioned. Dynamically parsing and mapping UDTs into Go structs.
Extending gocql’s marshaling/unmarshaling logic to handle UDTs seamlessly or do it separately to prevent backward compatibility issues. Also, we should be aware of synchronization issues like race conditions, stale data, etc.
I would say it is still relevant; maybe it should be provided to v2?

**Comment by joao-r-reis** [comment]
@jameshartig is this something that is related to the work you've been doing? I believe the answer is no but I'm not 100% sure.

Having this in 2.0 release would be nice but I don't think we need to break the API to do it so I'd actually prefer if we worked on it for a follow up 2.x minor release so we don't keep increasing the scope of 2.0.

**Comment by OleksiienkoMykyta** [comment]
> [@jameshartig](https://github.com/jameshartig) is this something that is related to the work you've been doing? I believe the answer is no but I'm not 100% sure.
> 
> Having this in 2.0 release would be nice but I don't think we need to break the API to do it so I'd actually prefer if we worked on it for a follow up 2.x minor release so we don't keep increasing the scope of 2.0.

Maybe we should somehow expose the scope, mark issues, or whatever? Because it's not clear what the high-priority issues are and what release issues belong to. I think it will help to handle all of these tasks later and not get lost in the mess.
Will be happy to help with it!

**Comment by jameshartig** [comment]
> @jameshartig is this something that is related to the work you've been doing? I believe the answer is no but I'm not 100% sure.

No, I didn't make any progress on this but it should be easier to do now after my changes since we could register them as specific types after loading them from the schema table.

> Having this in 2.0 release would be nice but I don't think we need to break the API to do it so I'd actually prefer if we worked on it for a follow up 2.x minor release so we don't keep increasing the scope of 2.0.

I agree, this seems like a major feature and I don't think we need to add even more to the release.

**Comment by joao-r-reis** [comment]
> Maybe we should somehow expose the scope, mark issues, or whatever? Because it's not clear what the high-priority issues are and what release issues belong to. I think it will help to handle all of these tasks later and not get lost in the mess. Will be happy to help with it!
 
I can create a `2.x` fixversion so that we can tag JIRAs for a follow up release after 2.0
